zebra-state/src/service/non_finalized_state.rs

```diff
@@ -167,6 +167,7 @@ impl NonFinalizedState {
             finalized_state.sapling_note_commitment_tree(),
             finalized_state.orchard_note_commitment_tree(),
             finalized_state.history_tree(),
+            finalized_state.current_value_pool(),
```

Comment:
Here we get the value pool of the finalized chain by using the function we previously added (in PR https://github.com/ZcashFoundation/zebra/pull/2599)